### PR TITLE
feat: Create test connection feature

### PR DESCRIPTION
**Pull request by ywei390**
## Description

Work Item ID: [AB#43208](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/43208)

Test connection example:

## How has it been tested? 
Manually in local

### CODE REVIEW

File: src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs [hunk 1]
```diff
@@ -456,6 +457,102 @@ public IPreviewImage GetPrimaryEntityPreviewImage(ExecutionContext context, IExt
...
+            var jobData = new GoogleMapsExternalSearchJobData(configDict);
+            var apiToken = jobData.ApiToken;
+            var client = new RestClient("https://maps.googleapis.com/maps/api");
+            var output = "json";
+            var placeDetailsEndpoint = $"place/details/{output}?";
+            var placeIdEndpoint = $"place/textsearch/{output}?";
+
+            var placeIdRequest = new RestRequest(placeIdEndpoint, Method.GET);
+            placeIdRequest.AddQueryParameter("key", apiToken);
+            placeIdRequest.AddQueryParameter("query", "Google" + " " + "1600 Amphitheatre Parkway, Mountain View, CA 94043.");
```
Comment (andrew-hardy) [inline src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs:472]: why `+ " " +` ?

File: src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs [hunk 2]
```diff
@@ -456,6 +457,102 @@ public IPreviewImage GetPrimaryEntityPreviewImage(ExecutionContext context, IExt
...
+
+                IRestResponse<CompanyDetailsResponse> response = null;
+
+                try
+                {
+                    response = client.ExecuteAsync<CompanyDetailsResponse>(request).Result;
+                }
+                catch (Exception exception)
+                {
+                    return new ConnectionVerificationResult(false, $"Could not return CompanyDetailsResponse from Google Maps. {exception}");
```
Comment (andrew-hardy) [inline src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs:506]: Update message

`$"Could not fetch company details from Google Maps. {exception.Message}"`

The above is now only putting the exception.Message in the response returned to the user as the user will not want to see a full stack trace. So make sure you also log the full exception

**Review by andrew-hardy — CHANGES_REQUESTED** [review]

File: src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs (hunk 1 above)
Comment (ywei390) [reply src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs]: removed

File: src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs (hunk 2 above)
Comment (ywei390) [reply src/ExternalSearch.Providers.GoogleMaps/GoogleMapsExternalSearchProvider.cs]: updated the message

**Review by andrew-hardy — APPROVED** [review]

**Pull request merged**